Python/traceback.c

```diff
@@ -1101,3 +1104,59 @@ _Py_DumpTracebackThreads(int fd, PyInterpreterState *interp,
     return NULL;
 }
 
+/* This is for faulthandler.
+ * Apparently, backtrace() doesn't play well across DLL boundaries on macOS */
+#if defined(HAVE_EXECINFO_H) && defined(HAVE_BACKTRACE) && defined(HAVE_BACKTRACE_SYMBOLS)
+void
+_Py_DumpStack(int fd)
+{
+#define BACKTRACE_SIZE 32
+#define TRACEBACK_ENTRY_MAX_SIZE 256
+    PUTS(fd, "Current thread's C stack trace (most recent call first):\n");
+    void *callstack[BACKTRACE_SIZE];
+    int frames = backtrace(callstack, BACKTRACE_SIZE);
+    if (frames == 0) {
+        // Some systems won't return anything for the stack trace
+        PUTS(fd, "  <system returned no stack trace>\n");
+        return;
+    }
+
+    char **strings = backtrace_symbols(callstack, BACKTRACE_SIZE);
+    if (strings == NULL) {
+        PUTS(fd, "  <not enough memory to get stack trace>\n");
+        return;
+    }
+    for (int i = 0; i < frames; ++i) {
+        char entry_str[TRACEBACK_ENTRY_MAX_SIZE];
+        snprintf(entry_str, TRACEBACK_ENTRY_MAX_SIZE, "  %s\n", strings[i]);
```

Comment:
Not sure if we do I in general, but you can suppress the return value of snprintf. I have been suppressing unused return values when used in a void function or if it's an inline call, but only for some function (ideally I'd like to do it everywhere but since we don't have a compiler warning because of this for now, I don't do it for old code that I'm not modifying).

OTOH, isn't the return value the number of characters that were written to the buffer? (excluding the null byte) why can't you use it instead of recomputing strlen?